service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/EnableAksContainerInsightsStep.java

```diff
@@ -0,0 +1,84 @@
+package bio.terra.landingzone.stairway.flight.create.resource.step;
+
+import bio.terra.landingzone.common.utils.HttpResponseUtils;
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.stairway.flight.utils.FlightUtils;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.containerservice.models.KubernetesCluster;
+import com.azure.resourcemanager.containerservice.models.ManagedClusterAddonProfile;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This step turns on Container insights (monitoring) for AKS by applying specific profile. We know
+ * that earlier ContainerInsight was created as a result of enabling monitoring feature in AKS. And
```

Comment:
I'm not able to follow lines 22-24 about ContainerInsight. It seems like you say that we can't use "useAADAuth=true", but then say that we should use "useAADAuth=true".

Reply:
Earlier when we used that option, we had problem with K8s monitoring because we turn on monitoring during K8s creation.  We need to use that option, but we need to enable K8s monitoring once K8s cluster is created.

Reply:
^ makes sense to me. Can you change the comment to say ^?

Reply:
The specific parts that I can't follow are lines 21-24 about the ContainerInsight.